src/widgets/dnshole/DnsHoleControls.tsx

```diff
@@ -0,0 +1,109 @@
+import { Badge, Button, Card, Group, Image, Stack, Text } from '@mantine/core';
+import { useTranslation } from 'next-i18next';
+import { IconDeviceGamepad, IconPlayerPlay, IconPlayerStop } from '@tabler/icons';
+import { useConfigContext } from '../../config/provider';
+import { defineWidget } from '../helper';
+import { WidgetLoading } from '../loading';
+import { IWidget } from '../widgets';
+import { useAdHoleControlMutation, useAdHoleSummeryQuery } from './query';
+import { PiholeApiSummaryType } from './type';
+
+const definition = defineWidget({
+  id: 'dns-hole-controls',
+  icon: IconDeviceGamepad,
+  options: {},
+  gridstack: {
+    minWidth: 3,
+    minHeight: 2,
+    maxWidth: 12,
+    maxHeight: 12,
+  },
+  component: AdHoleControlsWidgetTile,
+});
+
+export type IAdHoleControlsWidget = IWidget<(typeof definition)['id'], typeof definition>;
+
+interface AdHoleControlsWidgetProps {
+  widget: IAdHoleControlsWidget;
+}
+
+function AdHoleControlsWidgetTile({ widget }: AdHoleControlsWidgetProps) {
+  const { isInitialLoading, data, refetch } = useAdHoleSummeryQuery();
+  const { mutateAsync } = useAdHoleControlMutation();
+  const { t } = useTranslation('modules/dns-hole-controls');
+
+  const { config } = useConfigContext();
+
+  if (isInitialLoading || !data) {
+    return <WidgetLoading />;
+  }
+
+  return (
+    <Stack>
+      <Group grow>
+        <Button
+          onClick={async () => {
+            await mutateAsync('enabled');
+            await refetch();
```

Comment:
Please don't use refetch and more consider using query invalidation
https://tanstack.com/query/v4/docs/react/guides/query-invalidation